Description: Patch Android asset file handler to strip any instance of "./" from
the paths its given. The icon engine stuffs those in there unconditionally and
(reasonably) expects that to have no effect, but on Android's weird content URI
file system, it leads to the paths not resolving anymore.

--- a/src/plugins/platforms/android/qandroidassetsfileenginehandler.cpp
+++ b/src/plugins/platforms/android/qandroidassetsfileenginehandler.cpp
@@ -24,6 +24,7 @@
     if (file.startsWith(assetsPrefix))
         file.remove(0, prefixSize);
     file.replace("//"_L1, "/"_L1);
+    file.replace("/."_L1, ""_L1);
     if (file.startsWith(u'/'))
         file.remove(0, 1);
     if (file.endsWith(u'/'))
@@ -35,6 +36,7 @@
 {
     path = assetsPrefix + u'/' + path;
     path.replace("//"_L1, "/"_L1);
+    path.replace("/."_L1, ""_L1);
     return path;
 }
 
@@ -128,6 +130,7 @@
         }
         m_path = assetsPrefix + u'/' + m_path + u'/';
         m_path.replace("//"_L1, "/"_L1);
+        m_path.replace("/."_L1, ""_L1);
     }
 
     QString currentFileName() const
@@ -391,6 +394,7 @@
 
     QString path = fileName.mid(prefixSize);
     path.replace("//"_L1, "/"_L1);
+    path.replace("/."_L1, ""_L1);
     if (path.startsWith(u'/'))
         path.remove(0, 1);
     if (path.endsWith(u'/'))
